### DISCUSSION TITLE
Fill a pdf form tool

### DISCUSSION THREAD

**Discussion by Fahme** [human-tools/toolbox-web#13] [Ideas]
As a user it would be really cool to be able to have a tool that allows me to fill pdf forms easily with the same tools principals. Currently all of the existing softwares that i saw are expensive or uses servers to do the tasks. 

I'm thinking the human tool should be easy to use and fast. Also i'm not really sure if it's technically doable on client but would love to discuss further

Here's a premium website https://docspring.com which does the same thing.

**Comment by mkhatib** [comment]
I think that's a great idea and I am 90% sure it's doable. The library we're using [pdf-lib](https://pdf-lib.org.js) has a section on creating and filling forms. See:
```
Create new forms or fill and read existing fields. Check boxes, buttons, radio groups, dropdowns, option lists, and text fields are all supported.
```

So basically one would upload any PDF form they want to fill, and a human tool would parse it and present the forms in a more manageable format (web forms, styled, sectioned and step-by-step ...etc) and then users would download the original PDF with the forms filled in by the tool. I think this is what you were going for, correct?

**Reply by Fahme** [reply]
Yes! that's exactly what i mean, since the pdf-lib supports creating new forms i'm thinking the user can visually pinpoint areas on the document to add input fields and with the options of choosing fonts and colors maybe. 

This features opens alot of questions for discussion on how it should be implemented with the seamless UX because quite a  few softwares have it but not in the most performant and easy to use way. 

I'm hoping this tool can solve few of the issues i encounter in other softwares and not have to pay for anything :)

**Reply by mkhatib** [reply]
There are two separate things here to recognize:
1. A human has a PDF with an already build PDF form inside of it, like fields are in there they just might not be the easiest to fill so they upload them and we show the fields a bit more nicely formatted.
2. A human has a PDF with no actual PDF fields or mixed up fields and they want to mark areas they need to fill and then go to our human interface to fill it up.

I am not sure if you were talking about only 2 or not. Or you don't see that 1. is an actual problem because fields are already available and users are able to fill them in the PDF itself. What do you think?

**Reply by Fahme** [reply]
I was aiming for point 2 more than 1, as filling out a pdf which has inputs in it is already easy. 

But i believe combining both would defiantly be better, because there's times where a user would want to mark areas that might not have inputs to fill, also we can have more consistency in term of fonts, colors etc .. in both scenarios.

**Reply by mkhatib** [reply]
Hey! I have pushed the new Quick PDF Sign tool that now allows people to add text and drawings to the PDF. So we can add some more items to it.
